lusidtools/cocoon/utilities.py

```diff
@@ -1327,3 +1328,72 @@ def update_value(d: typing.Union[dict, str], val: typing.Union[str, float]):
     d = val
 
     return d
+
+
+def group_request_into_one(
+    model_type, request_list: list, attribute_for_grouping: list, batch_index=0
+):
+
+    """
+    Description
+    ------------
+    This function take a list of requests and collates an attribute from each request, adding the collated attributes
+    back onto the first request in the list. The function returns the modified first request.
+    For example, the function can take a list of CreatePortfolioGroupRequests, extract the "values" or portfolios from
+    each request, and then add all portfolios back onto the first request in the list.
+    :param lusid.models model_type:
+    :param list request_list:
+    :param list attribute_for_grouping:
+    :return: a single LUSID request
+    """
+
+    # Define a base request for modifying - this is the first request in the list by default
+
+    base_request = request_list[batch_index]
+
+    for attrib in attribute_for_grouping:
+
+        if getattr(models, model_type).openapi_types[attrib] == "list[ResourceId]":
```

Comment:
I'd update this to just check if "list[" in getattr(models, model_type).openapi_types[attrib] so that this will work for lists of things other than ResourceIds